src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/steps/data/merge-form-data.ts

```diff
@@ -0,0 +1,97 @@
+import { JsonFormData } from "../../add-polling-source-form.types";
+
+export const MERGE_FORM_DATA: JsonFormData = {
+    append: {
+        controls: [],
+    },
+    ledger: {
+        controls: [
+            {
+                name: "primaryKey",
+                label: "Primary keys",
+                value: "",
+                type: "array-key",
+                placeholder: "Enter primary key...",
+                tooltip:
+                    "Names of the columns that uniquely identify the record throughout its lifetime",
+                options: {
+                    formArrayName: "primaryKey",
+                    buttonText: "+ Add primary key",
+                },
+                validators: {},
+            },
```

Comment:
I think that fields like these, where we select column names, should have a better validation logic. i.e. we cannot use spaces in column names, but the form allows it:

![image](https://user-images.githubusercontent.com/8875909/233832300-08e08d1a-345d-4ad8-ac79-4cdf92438887.png)

Reply:
Trim spaces in this field